src/Microsoft.ComponentDetection.Contracts/TypedComponent/CondaLockComponent.cs

```diff
@@ -0,0 +1,47 @@
+namespace Microsoft.ComponentDetection.Contracts.TypedComponent;
+
+using System.Collections.Generic;
+
+public class CondaLockComponent : TypedComponent
+{
+    public CondaLockComponent(string name, string version, string category, Dictionary<string, string> dependencies, Dictionary<string, string> hash, string manager, bool optional, string platform, string url)
+    {
+        this.Name = this.ValidateRequiredInput(name, nameof(this.Name), nameof(ComponentType.Conda));
+        this.Version = this.ValidateRequiredInput(version, nameof(this.Version), nameof(ComponentType.Conda));
+        this.Category = category;
+        this.Dependencies = dependencies;
+        this.Hash = hash;
+        this.Manager = manager;
+        this.Optional = optional;
+        this.Platform = platform;
+        this.Url = url;
+        this.Version = version;
```

Comment:
This is not necessary, you are already setting the validated value in line 10.
```suggestion
```

Reply:
Great call out! 

As discussed offline also, we don't actually need all of these information for component governance. That's why I refactored the code to only use the minimal set of required properties (the same way as the PipComponent works). This especially has the benefit that we will avoid adding duplicate packages. The previous implementation detected 215 Conda packages in the verification tests. A lot of them have been duplicates. Now, with the new implementation we are down to 88 packages in the verification list. This is the number of deduplicate packages in the conda-lock file.